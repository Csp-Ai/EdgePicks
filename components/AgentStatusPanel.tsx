import React, { useState } from 'react';
import { agents as agentRegistry } from '../lib/agents/registry';
import { formatAgentName } from '../lib/utils';
import type { AgentLifecycle, AgentName } from '../lib/types';

export type AgentStatusMap = Record<
  AgentName,
  { status: AgentLifecycle['status'] | 'idle'; durationMs?: number }
>;

interface Props {
  statuses: Partial<AgentStatusMap>;
}

const AgentStatusPanel: React.FC<Props> = ({ statuses }) => {
  const [open, setOpen] = useState(false);

  return (
    <div className="fixed left-0 right-0 bottom-16 mx-auto max-w-md">
      <div className="bg-white border rounded-t shadow-lg">
        <button
          className="w-full px-4 py-2 text-left font-medium bg-gray-100"
          onClick={() => setOpen((o) => !o)}
        >
          {open ? 'Hide Agent Status' : 'Show Agent Status'}
        </button>
        {open && (
          <ul className="divide-y">
            {agentRegistry.map(({ name }) => {
              const info = statuses[name];
              let label = 'Waiting';
              if (info?.status === 'started') label = 'Runningâ€¦';
              else if (info?.status === 'completed')
                label = `Completed in ${info.durationMs ?? 0}ms`;
              else if (info?.status === 'errored') label = 'Error';
              return (
                <li
                  key={name}
                  className="flex items-center justify-between px-4 py-2 text-sm"
                >
                  <span>{formatAgentName(name)}</span>
                  <span>{label}</span>
                </li>
              );
            })}
          </ul>
        )}
      </div>
    </div>
  );
};

export default AgentStatusPanel;
