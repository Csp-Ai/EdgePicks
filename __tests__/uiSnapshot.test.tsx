import { render } from '@testing-library/react';
import React from 'react';

jest.mock('../lib/agents/registry', () => ({
  agents: [
    { name: 'injuryScout', description: '', type: 'injury', weight: 0.5, sources: [] },
  ],
}));

import AgentStatusPanel from '../components/AgentStatusPanel';

describe('AgentStatusPanel UI', () => {
  it('renders default state', () => {
    const { asFragment } = render(
      <AgentStatusPanel statuses={{}} sessionId="test-session" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
