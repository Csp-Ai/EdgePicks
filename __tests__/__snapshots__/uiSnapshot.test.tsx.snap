// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentStatusPanel UI renders default state 1`] = `
<DocumentFragment>
  <div
    class="fixed left-0 right-0 bottom-16 mx-auto max-w-md"
  >
    <div
      class="bg-white border rounded-t shadow-lg"
    >
      <button
        aria-expanded="false"
        class="w-full px-4 py-2 text-left font-medium bg-gray-100 focus:outline-none focus:ring-2 focus:ring-blue-500"
      >
        Show Agent Status
      </button>
    </div>
  </div>
</DocumentFragment>
`;
